File: Array/Smallest Element In Array/smallestElementInArray.cpp
/******************************************************************************
            Program to compute the smallest element in array
******************************************************************************/
#include<iostream>
#include<cstdio>
#define maxSize 100
using namespace std;

int main()
{
    int array[maxSize], size;
    int min;

    // Scanning the size of the array
    cout << "Enter the size of the array: ";
    cin >> size;

    // Scanning the elements of the array
    cout << "Enter the " << size << " elements of the array: " << endl;
    for(int i=0; i<size; i++)
    {
        printf("ELement [%d]: ",i);
        cin >> array[i];
    }

    // Printing the elements of the array
    cout << "The input array: " << endl;
    for(int i=0; i<size; i++)
    {
        cout << array[i] << " ";
    }

    // Assigning the first element of the array to min variable
    min = array[0];

    // Checking for elements minimum than value of min variable
    for(int i=0; i<size; i++)
    {
        if(array[i] < min)
        {
            min = array[i];
        }
    }

    // Printing out the result
    cout << "\nThe smallest element of input array: " << min << endl;

    return 0;
}

File: Array/Smallest Element In Array/smallestElementInArray.java
import java.util.*;

public class smallestElementInArray 
{
    public static void main(String[] args)
    {
        try(Scanner sc = new Scanner(System.in))
        {
            // Scanning the size of the array
            System.out.print("Enter the size of the array: ");
            int size = sc.nextInt();

            int array[] = new int[size];
            // Scanning the elements of array
            System.out.println("Enter the elements of array: ");
            for(int i=0; i<size; i++)
            {
                System.out.print("Element ["+ i +"]: ");
                array[i] = sc.nextInt();
            }

            //Printing the array
            System.out.println("Array: ");
            for(int i=0; i<size; i++)
            {
                System.out.print(array[i] + " ");
            }
            System.out.println();

            // Assigning the first element of array to min variable
            int min = array[0];

            // Computing smallest element in array
            for(int i=0; i<size; i++)
            {
                if(array[i] < min)
                {
                    min = array[i];
                }
            }

            // Printing the result
            System.out.println("The smallest element of the array: "+min);
        }
    }
}
